### PR TITLE
Ft implement monaco editor 166702872

### PR DESCRIPTION
**Pull request by shegsteham**
#### What does this PR do?
Switches playground editor from Code Mirror Editor to Monaco editor

#### Description of Task to be completed
- add monaco-editor module
- initialise Monaco editor in `src/commons/js/monaco-init.js`
- add Monaco language file in `src/commons/js/monaco-lang.js`
- update `src/playground/js/playground.js`
- update `src/dash/js/spec.js`

#### How should this be manually tested?
- switch to the branch `ft-implement-monaco-editor-166702872`
- start the server using `yarn develop-playground`
- navigate to `http://localhost:1234/jqe3zYO8xTfiuCLDEEgu` on the browser
- you will notice the Monaco editor with prepopulated codes

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#166702872](https://www.pivotaltracker.com/story/show/166702872)
[#166771642](https://www.pivotaltracker.com/story/show/166771642)

#### Screenshots (if appropriate)
image
image
image
image

### CODE REVIEW

File: src/playground/css/main.scss [hunk 1]
```diff
@@ -167,6 +168,11 @@ body[data-assessment] header.mdc-top-app-bar--short-collapsed button {
   min-height: 100%;
 }
 
+// hide cut/paste and command palate from the context menu options
+li.action-item:nth-child(n + 6) {
```
Comment (charlesodili) [inline src/playground/css/main.scss]: Can you better target the LI element here. E.g Isn't it a descendant of some monaco editor element with CSS classes ?

File: src/commons/js/monacoEditor/monaco-lang.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  r: 'r',
+  ruby: 'ruby',
+  solidity: 'solidity',
+  swift: 'swift',
+  typescript: 'typescript',
+  xml: 'xml',
+  yaml: 'yaml',
+  css: 'css',
+  scss: 'scss'
+};
```
[the remark this answers is not in the document]
Comment (shegsteham) [reply src/commons/js/monacoEditor/monaco-lang.js]: @charlesodili, I just added the relevant languages while envisaging future support by Gradr. However, I will reduce it to just the ones required for now. It has no performance impact.

File: src/playground/css/main.scss (hunk 1 above)
Comment (shegsteham) [reply src/playground/css/main.scss]: @chalu, you are right, but the challenge I encounter is that; there are about 6 descendants in the context menu and they all have the same class names. Hence it was difficult to modify the cut, copy and command palate options using classNames. I.E it affects all the descendants

**Comment by charlesodili** [comment]
What will it take to add linting in the Playground editor, or is it there and I can't see it? Like having the editor display an error marker as soon as I type invalid HTML, JS, or CSS. Will this only work when we have single language editors (i.e multi-file support) instead of now that a single HTML file contains HTML, CSS and JS

**Comment by shegsteham** [comment]
@charlesodili, it is possible to integrate linting using Jslint or Eslint, but I will suggest we implement that on a separate story after implementing the multi-tabs

**Pull request merged**